frontends/web/src/routes/device/bitbox02/setup/wallet-create.tsx

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import * as bitbox02 from '../../../../api/bitbox02';
+import { alertUser } from '../../../../components/alert/Alert';
+import { Wait } from './wait';
+import { ChecklistWalletCreate } from './checklist';
+import { SetDeviceName } from './name';
+import { SetPassword } from './password';
+
+type TCreateWalletStatus = 'intro' | 'setPassword' | 'createBackup';
+
+type TWait = {
+  title: string;
+  text?: string;
+};
+
+type Props = {
+  deviceID: string;
+  ensureSDCard: () => Promise<boolean>;
+  isSeeded: boolean;
+  onAbort: () => void;
+};
+
+export const CreateWallet = ({
+  deviceID,
+  ensureSDCard,
+  isSeeded,
+  onAbort,
+}: Props) => {
+  const { t } = useTranslation();
+  const [status, setStatus] = useState<TCreateWalletStatus>('intro');
+  const [errorText, setErrorText] = useState('');
+  const [waitView, setWaitView] = useState<TWait>();
+  const [hasSDCard, setSDCard] = useState<boolean>();
+
+  useEffect(() => {
+    bitbox02.checkSDCard(deviceID).then(setSDCard);
+  }, [deviceID]);
+
+  const ensurePassword = async () => {
+    setStatus('setPassword');
+    try {
+      const result = await bitbox02.setPassword(deviceID);
+      if (!result.success) {
+        if (result.code === bitbox02.errUserAbort) {
+          // On user abort, just go back to the first screen. This is a bit lazy, as we should show
+          // a screen to ask the user to go back or try again.
+          setErrorText('');
+          onAbort();
+        } else {
+          setErrorText(t('bitbox02Wizard.noPasswordMatch'));
+          ensurePassword();
+        }
+        // show error and do NOT continue to createBackup
+        return;
+      }
+      setErrorText('');
+      setStatus('createBackup');
```

Comment:
this is the state during create wallet moved from bitbox02 to wallet-create.tsx component see this comment https://github.com/digitalbitbox/bitbox-wallet-app/pull/2120/files#r1202263901

Reply:
I mean that 'createBackup' status is not actually used any more, right? The `ChecklistWalletCreate` component depends only on the device being seeded.

Reply:
thank you, I forgot I added it back so that it is the same as in the old implementation https://github.com/digitalbitbox/bitbox-wallet-app/pull/2120/commits/400915115855c0a202ace731d6011a4f0118e9f9